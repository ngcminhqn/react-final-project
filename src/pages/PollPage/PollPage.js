import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { usersActions } from "../../features/users/usersSlice";
import {
  AvatarStyled,
  ButtonClickContainerStyled,
  ButtonClickStyled,
  PollContainerStyled,
  TitleStyled,
} from "./PollPage.styled";
import { Card, Col, Row } from "antd";
import { useParams } from "react-router-dom";
import { questionsActions } from "../../features/questions/questionsSlice";
import { EMPTY_STRING } from "../../constants/common";
import { getUser } from "../../helpers/user";
import Loading from "../../components/Loading/Loading";

const PollPage = () => {
  const dispatch = useDispatch();
  const { questionId } = useParams();

  const { loading, questionById } = useSelector((state) => state.questions);
  const user = getUser();
  const { users } = useSelector((state) => state.users);

  useEffect(() => {
    if (questionId) {
      const payload = {
        questionId: questionId,
      };
      dispatch(
        questionsActions.handleGetQuestionById({
          data: payload,
        })
      );
    }
  }, [dispatch, questionId]);

  useEffect(() => {
    dispatch(usersActions.handleGetListUsers());
  }, [dispatch]);

  const saveQuestionByAnswer = (type) => {
    const payload = {
      authedUser: user?.id,
      questionId: questionById?.id,
      answer: type,
    };
    if (user?.id && questionById?.id) {
      dispatch(
        questionsActions.handleSaveQuestionByAnswer({
          data: payload,
        })
      );
    }
  };

  return (
    <PollContainerStyled>
      <Loading isLoading={loading} />
      <TitleStyled>Poll by {questionById?.author}</TitleStyled>
      {questionById && users[questionById.author] ? (
        <AvatarStyled
          src={users[questionById?.author ?? ""].avatarURL ?? undefined}
          alt={questionById?.author || EMPTY_STRING}
        />
      ) : null}
      <TitleStyled>Would You Rather</TitleStyled>
      <Row gutter={8} style={{ width: "80%" }}>
        <Col span={12}>
          <Card style={{ textAlign: "center" }}>
            {questionById?.optionOne?.text || EMPTY_STRING}
            <ButtonClickContainerStyled>
              <ButtonClickStyled
                onClick={() => saveQuestionByAnswer("optionOne")}
                type="primary"
                block
              >
                Click
              </ButtonClickStyled>
            </ButtonClickContainerStyled>
          </Card>
        </Col>

        <Col span={12}>
          <Card style={{ textAlign: "center" }}>
            {questionById?.optionTwo?.text || EMPTY_STRING}
            <ButtonClickContainerStyled>
              <ButtonClickStyled
                onClick={() => saveQuestionByAnswer("optionTwo")}
                type="primary"
                block
              >
                Click
              </ButtonClickStyled>
            </ButtonClickContainerStyled>
          </Card>
        </Col>
      </Row>
    </PollContainerStyled>
  );
};

export default PollPage;
