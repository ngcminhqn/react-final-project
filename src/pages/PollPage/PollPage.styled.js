import { Button } from "antd";
import styled from "styled-components";

export const PollContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 100vh;
  padding: 20px 0;
`;

export const TitleStyled = styled.h1``;

export const ButtonClickContainerStyled = styled.div`
  margin: 10px -24px -24px -24px;
  border-top: 1px solid #f0f0f0;
`;

export const ButtonClickStyled = styled(Button)`
  border-top-left-radius: 0px;
  border-top-right-radius: 0px;
  background-color: #00b3b3;

  &:hover {
    background-color: #00b3b3 !important;
  }
`;

export const AvatarStyled = styled.img`
  width: 300px;
  height: 300px;
`;
