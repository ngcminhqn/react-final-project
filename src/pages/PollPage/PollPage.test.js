import PollPage from "./PollPage";
import { renderWithProviders } from "../../utils/test-utils";

describe("PollPage", () => {
  it("render PollPage", async () => {
    const { container } = renderWithProviders(<PollPage />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
