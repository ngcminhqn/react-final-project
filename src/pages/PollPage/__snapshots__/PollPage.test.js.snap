// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollPage render PollPage 1`] = `
<div>
  <div
    class="sc-aYaIB fxIVEF"
  >
    <h1
      class="sc-gEvDqW"
    >
      Poll by 
    </h1>
    <h1
      class="sc-gEvDqW"
    >
      Would You Rather
    </h1>
    <div
      class="ant-row css-dev-only-do-not-override-m4timi"
      style="margin-left: -4px; margin-right: -4px; width: 80%;"
    >
      <div
        class="ant-col ant-col-12 css-dev-only-do-not-override-m4timi"
        style="padding-left: 4px; padding-right: 4px;"
      >
        <div
          class="ant-card ant-card-bordered css-dev-only-do-not-override-m4timi"
          style="text-align: center;"
        >
          <div
            class="ant-card-body"
          >
            -
            <div
              class="sc-eqUzNf kAyPmj"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-m4timi ant-btn-primary ant-btn-block sc-fqkwJk dPkbA-D"
                type="button"
              >
                <span>
                  Click
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-col ant-col-12 css-dev-only-do-not-override-m4timi"
        style="padding-left: 4px; padding-right: 4px;"
      >
        <div
          class="ant-card ant-card-bordered css-dev-only-do-not-override-m4timi"
          style="text-align: center;"
        >
          <div
            class="ant-card-body"
          >
            -
            <div
              class="sc-eqUzNf kAyPmj"
            >
              <button
                class="ant-btn css-dev-only-do-not-override-m4timi ant-btn-primary ant-btn-block sc-fqkwJk dPkbA-D"
                type="button"
              >
                <span>
                  Click
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
